jwst/extract_1d/extract.py

```diff
@@ -539,31 +539,34 @@ def update_from_shape(ap, im_shape):
     ystop = ap.ystop
 
     truncated = False
+
     if ap.xstart < 0:
         xstart = 0
         truncated = True
+
     if ap.xstop >= nx:
-        xstop = nx - 1                          # limits are inclusive
+        xstop = nx - 1  # limits are inclusive
         truncated = True
+
     if ap.ystart < 0:
         ystart = 0
         truncated = True
+
     if ap.ystop >= ny:
         ystop = ny - 1
         truncated = True
 
-    ap_shape = Aperture(xstart=xstart, xstop=xstop,
-                        ystart=ystart, ystop=ystop)
+    ap_shape = Aperture(xstart=xstart, xstop=xstop, ystart=ystart, ystop=ystop)
 
-    return (ap_shape, truncated)
+    return ap_shape, truncated
 
 
-def aperture_from_wcs(wcs, verbose):
+def aperture_from_wcs(wcs: WCS, verbose: bool) -> Union[NamedTuple, None]:
     """Get the limits over which the WCS is defined.
 
     Parameters
     ----------
-    wcs : data model
+    wcs : data model ?
```

Comment:
I'm guessing it's not a data model, but a WCS object.

Reply:
I was confused on what the object was supposed to be since it says `wcs` but then describes the parameter as a "data model," so I was hoping that somebody might have a bit more contextual info here, but I think I agree with your guess!